src/Raven.Client/Documents/Queries/Facets/RangeFacet.cs

```diff
@@ -269,16 +265,19 @@ private static string GetStringRepresentation(string fieldName, ExpressionType l
             if (lValue != null && rValue != null)
             {
                 if (leftOp == ExpressionType.GreaterThanOrEqual && rightOp == ExpressionType.LessThanOrEqual)
-                    return $"{fieldName} between {GetStringValue(lValue)} and {GetStringValue(rValue)}";
+                    return $"{fieldName} between {GetStringValue(lValue, addQueryParameter)} and {GetStringValue(rValue, addQueryParameter)}";
 
-                return $"{GetStringRepresentation(fieldName, leftOp, lValue)} and {GetStringRepresentation(fieldName, rightOp, rValue)}";
+                return $"{GetStringRepresentation(fieldName, leftOp, lValue, addQueryParameter)} and {GetStringRepresentation(fieldName, rightOp, rValue, addQueryParameter)}";
             }
 
             throw new InvalidOperationException("Unable to parse the given operation into a facet range!!! ");
         }
 
-        private static string GetStringValue(object value)
+        private static string GetStringValue(object value, Func<object, string> addQueryParameter)
         {
+            if (addQueryParameter != null && (value is DateTime || value is DateTimeOffset || value is string))
+                return "$" + addQueryParameter(value);
+
             switch (value.GetType().FullName)
             {
                 //The nullable stuff here it a bit weird, but it helps with trying to cast Value types
```

Comment:
Need to handle the nullable stuff here as well.
Also, why are we doing string comparisons for the types?

Reply:
Let's handle this everything as a parameters, not just dates & strings

Reply:
String comparision is a left-over from 3.x. I did not notice this when I was implementing Facets in 4.0